src/cycax/cycad/cycad_part.py

```diff
@@ -256,50 +282,47 @@ def insert_hole(self, hole: Holes):
 
         self.features.append(hole)
         self.move_holes.append(hole)
-
-    def side_location_calculator(self, side: str, x: float, y: float, sink: float = 0.0) -> tuple:
+        
+    def save(self):
         """
-        Calculate given the side of and object and the relative x and y location on that side where the definite location is.
-        Args:
-            x: x location
-            y: y location
-            side: side of object
-
-        Raises:
-            ValueError: if the side is not one of the available and expected sides.
+        This takes the provided part and will create its dictionary and export it to a json
+        """
+        dir_name = f"{os.getcwd()}/{self.part_no}/{self.part_no}"
```

Comment:
Probably just one (self.part_no} is needed.

FYI: Good coding practice would be to use os.path.join and not string concatenate with "/" as separators, you might break windows. bleee to bad.

But have a look at [pathlib](https://docs.python.org/3/library/pathlib.html) in Python standard library it is a lot better to use than os. Take a moment and read up on pathlib it is worth the time.
e.g. Below should be the same.
```
from pathlib import Path # Import at the top.
dir_name = Path('.') / self.part_no # Yes they overloaded device to be concat for paths. Pretty cool.
dir_name.mkdir(exist_ok=True)
file_path = dir_name / f"{self.part_no}.json"
json.dump(self.export(), file_path.open("w+"))
```